### ISSUE TITLE
[Bug]: The 'video.api.livestream/events' channel sent a message from native to Flutter on a non-platform thread

### ISSUE THREAD

**Issue by imsujan276** [apivideo/api.video-flutter-live-stream#47]
### Version

v1.1.1

### Which operating systems have you used?

- [ ] Android
- [X] iOS

### Environment that reproduces the issue

* iOS simulator iPhone 15 Pro max

### Is it reproducible in the example application?

Yes

### RTMP Server

api.video

### Reproduction steps

1. start the streaming in ios
2. stop it 
3. you can see the error message in the console

### Expected result

There should not be any error message in the console and the package should have sent the Platform channel messages on the platform thread

### Actual result

The app give the following error message in console 

[ERROR:flutter/shell/common/shell.cc(1015)] The 'video.api.livestream/events' channel sent a message from native to Flutter on a non-platform thread. Platform channel messages must be sent on the platform thread. Failure to do so may result in data loss or crashes, and must be fixed in the plugin or application code creating that channel.
See https://docs.flutter.dev/platform-integration/platform-channels#channels-and-platform-threading for more information.

### Additional context

Example:

``` dart
ApiVideoLiveStreamController createLiveStreamController() {
    return ApiVideoLiveStreamController(
      initialAudioConfig: rtmsConfigaudioConfig,
      initialVideoConfig: rtmsConfig.videoConfig,
      onConnectionSuccess: () {
        log('VideoStream: Connection succeeded');
      },
      onConnectionFailed: (error) {
        log('VideoStream: Connection failed: $error');
      },
      onDisconnection: () {
        log('VideoStream: Connection Disconnected');
      },
      onError: (err) {
        log('VideoStream: Connection Error: $err');
      },
    );
  }
```

### Relevant logs output

```shell
[log] VideoStream: Stream Started
[ERROR:flutter/shell/common/shell.cc(1015)] The 'video.api.livestream/events' channel sent a message from native to Flutter on a non-platform thread. Platform channel messages must be sent on the platform thread. Failure to do so may result in data loss or crashes, and must be fixed in the plugin or application code creating that channel.
See https://docs.flutter.dev/platform-integration/platform-channels#channels-and-platform-threading for more information.
[log] VideoStream: Connection succeeded
```

**Comment by ThibaultBee** [comment]
Could you test branch with fix https://github.com/apivideo/api.video-flutter-live-stream/tree/fix/ios_event_thread?

**Comment by imsujan276** [comment]
But now it takes several seconds to to get the onConnectionSuccess callback. 

Previously, it happens instantly

**Comment by ThibaultBee** [comment]
Hi,
I made a fix in the iOS native RTMP lib: https://github.com/apivideo/api.video-swift-live-stream
I will update this dependency before releasing a new version of `apivideo_live_stream`.

**Comment by imsujan276** [comment]
Thank you

I will keep an eye on the update

**Comment by ThibaultBee** [comment]
Could you test on the `main` branch?

**Comment by imsujan276** [comment]
yeah but does it include the changes made for [Issue 48](https://github.com/apivideo/api.video-flutter-live-stream/issues/48) for Android too?

**Comment by ThibaultBee** [comment]
yes

**Comment by imsujan276** [comment]
i tried it using the main branch, it is still taking several seconds for the stream connection to succeed.

**Comment by imsujan276** [comment]
https://github.com/apivideo/api.video-flutter-live-stream/assets/11173809/c980fab2-719b-4a50-a473-f35992de4177

You can see in the video, it takes around 10 seconds for the connection to succeed.

**Comment by ThibaultBee** [comment]
🤔 
- Do you have the same issue with the example?
- If yes, do you have the example from https://github.com/apivideo/api.video-swift-live-stream?

**Comment by imsujan276** [comment]
1. Yes i have the same issue while running the example in iOS too
2. NO, I don't have an example from that URL

**Comment by ThibaultBee** [comment]
Sorry, could you test with the example from https://github.com/apivideo/api.video-swift-live-stream?

**Comment by imsujan276** [comment]
I think that repo is for native iOS app. I am using flutter so not sure how

**Comment by ThibaultBee** [comment]
Do you have xcode? I guess yet because you need it for Flutter.
Run `open Examples/ExampleUIKit/ExampleUIKit.xcodeproj/`

If I ask so, it just to know if the issue comes from the Flutter iOS part or the native iOS part.

**Comment by imsujan276** [comment]
It worked really fast. No issues

**Comment by ThibaultBee** [comment]
I did have a look at Flutter iOS code but could not find anything.
Are you sure to be up to date? (last version of the internal library with `pod update`,...)

**Comment by imsujan276** [comment]
Yes.

I did pull the latest branch from git and run pod update, pod install --repo-update. 

Installed and run correctly. just giving me that 10 seconds delay to succeed the connection

**Comment by ThibaultBee** [comment]
The original issue is solved in v1.2.0. Please reopen it if you feel something is missing.

About your connection issue, as it is not related to the opened issue. Please open a new issue.
